src/verify.c

```diff
@@ -348,125 +348,128 @@ static int get_all_files(struct manifest *official_manifest, struct list *subs)
 	printf("\n");
-	if (isfailed) {
+	if (free_list) {
 		list_free_list(files);
 	}
 
 	return end_full_download();
 }
 
-static int get_missing_files(struct manifest *official_manifest)
+static int get_missing_files(struct list *files)
 {
 	int ret;
-	struct list *failed = NULL;
 	int retries = 0;  /* We only want to go through the download loop once */
 	int timeout = 10; /* Amount of seconds for first download retry */
 
-RETRY_DOWNLOADS:
-	/* when fixing (not installing): queue download and mark any files
-	 * which are already verified OK */
-	ret = start_full_download(true);
-	if (ret != 0) {
-		/* If we hit this point, the network is accessible but we were
-		 * 	unable to download the needed files. This is a terminal error
-		 * 	and we need good logging */
-		fprintf(stderr, "Error: Unable to download neccessary files for this OS release\n");
-		return -EFULLDOWNLOAD;
-	}
-
-	if (failed != NULL) {
-		failed = download_loop(failed, 1);
-	} else {
-		failed = download_loop(official_manifest->files, 0);
-	}
+	while (files) {
+		ret = start_full_download(true);
+		if (ret != 0) {
+			/* If we hit this point, the network is accessible but we were
+			 * unable to download the needed files. This is a terminal error
+			 * and we need good logging */
+			goto error;
+		}
 
-	/* Set retries only if failed downloads exist, and only retry a fixed
-	   amount of &times */
-	if (list_head(failed) != NULL && retries < MAX_TRIES) {
-		increment_retries(&retries, &timeout);
-		fprintf(stderr, "Starting download retry #%d\n", retries);
-		clean_curl_multi_queue();
-		goto RETRY_DOWNLOADS;
-	}
+		files = download_loop(files, retries > 0);
 
-	if (retries >= MAX_TRIES) {
-		fprintf(stderr, "ERROR: Could not download all files, aborting update\n");
-		list_free_list(failed);
-		return -EFULLDOWNLOAD;
+		/* Set retries only if failed downloads exist, and only retry a fixed
+		   amount of &times */
+		if (list_head(files)) {
+			if (retries >= MAX_TRIES) {
+				list_free_list(files);
+				goto error;
+			}
+			increment_retries(&retries, &timeout);
+			fprintf(stderr, "Starting download retry #%d\n", retries);
+			clean_curl_multi_queue();
+		}
 	}
 
 	return 0;
+
+error:
+	fprintf(stderr, "Error: Unable to download neccessary files for this OS release\n");
+	return -EFULLDOWNLOAD;
+}
+
+/*
+ * Check if the hash of all files in the list matches the system and in this case mark
+ * them as do_not_update.
+ * If cmdline_option_quick is set, use lazy hash mode and just check if file exists.
+ */
+static void check_files_hash(struct list *files)
+{
+	struct list *iter;
+	unsigned int complete = 0;
+	unsigned int total = list_len(files);
+
+	fprintf(stderr, "Verifying files\n");
+	iter = list_head(files);
+	while (iter) {
+		struct file *f = iter->data;
+		char *fullname;
+		bool valid;
+
+		print_progress(complete, total);
+		complete++;
+		iter = iter->next;
+		if (f->is_deleted || f->do_not_update)
```

Comment:
I prefer to see brackets even on one-line ifs. I notice you do use brackets elsewhere and the rest of the codebase should be as well.

Reply:
my bad